ucgenerator/main.go

```diff
@@ -0,0 +1,598 @@
+		jen.Id("o").Op("*"+o.dtoStructName),
+	).Id("*" + o.tfoStructName).Block(body...)
+}
+
+// genAttrsMap creates attributes map for Flatten functions to "unwrap" response json into TF object
+func genAttrsMap(f *jen.File, o *object) {
+	values := jen.Dict{}
+	for _, p := range o.properties {
+		key := jen.Lit(p.tfName)
+		switch p.Type {
+		case objectTypeArray, objectTypeObject:
+			var v jen.Code
+			if p.isNestedBlock() {
+				v = jen.Qual(importTypes, "ObjectType").Values(jen.Dict{
+					jen.Id("AttrTypes"): jen.Id(p.attrsName),
+				})
+			} else {
+				v = jen.Qual(importTypes, getTFType(p.ArrayItems)+"Type")
+			}
+			values[key] = jen.Qual(importTypes, "ListType").Values(jen.Dict{jen.Id("ElemType"): v})
+		default:
+			values[key] = jen.Qual(importTypes, getTFType(p)+"Type")
+		}
+	}
+	f.Var().Id(o.attrsName).Op("=").Map(jen.String()).Qual(importAttr, "Type").Values(values)
+}
+
+// genTFObject creates TF object (for plan)
+func genTFObject(f *jen.File, o *object) {
+	fields := make([]jen.Code, 0)
+	for _, p := range o.properties {
+		fields = append(fields, jen.Id(p.camelName).Qual(importTypes, getTFType(p)).Tag(map[string]string{"tfsdk": p.tfName}))
+	}
+	f.Comment(fmt.Sprintf("%s %s", o.tfoStructName, getDescription(o)))
+	f.Type().Id(o.tfoStructName).Struct(fields...)
+}
+
+// genDTOObject creates DTO object to send over HTTP
+func genDTOObject(f *jen.File, o *object) {
+	fields := make([]jen.Code, 0)
+	for _, p := range o.properties {
+		tags := map[string]string{"json": p.jsonName, "groups": "create"}
+		if !p.Required {
+			tags["json"] += ",omitempty"
+		}
+		if !p.CreateOnly {
+			tags["groups"] += ",update"
+		}
+		fields = append(fields, jen.Id(p.camelName).Id(getDTOType(p)).Tag(tags))
+	}
+	f.Comment(o.dtoStructName + " request/response object")
+	f.Type().Id(o.dtoStructName).Struct(fields...)
+}
+
+// genSchema generates TF schema. For root object only, i.e. RedisUserConfig
+func genSchema(f *jen.File, o *object, name, pkg string) {
+	if !o.isRoot {
+		return
+	}
+
+	funcName := fmt.Sprintf("New%sSchema", name)
+	f.Comment(fmt.Sprintf("%s returns %s schema", funcName, strings.ToLower(name)))
+	f.Func().Id(funcName).Params().Qual(pkg, "ListNestedBlock").Block(
+		jen.Return(getSchemaAttributes(o, pkg)),
+	)
+}
+
+func getSchemaAttributes(o *object, pkg string) jen.Code {
+	isResource := pkg == importResourceSchema
+	blocks := jen.Dict{}
+	attribs := jen.Dict{}
+
+	// Array properties are its item properties
+	properties := o.properties
+	if o.Type == objectTypeArray {
+		properties = o.ArrayItems.properties
+	}
+
+	for _, p := range properties {
+		key := jen.Lit(p.tfName)
+		if p.isNestedBlock() {
+			blocks[key] = getSchemaAttributes(p, pkg)
+		} else {
+			// For scalars
+			var value *jen.Statement
+			switch p.Type {
+			case objectTypeObject:
+				// Schemaless map
+				//value = jen.Qual(importTypes, "StringType")
+				panic("schemaless objects are not supported")
```

Comment:
Can we add error handling here and in a few other places where we use panic?  It would be more predictable if we did not through panic in the middle of the business logic.

Reply:
> Can we add error handling here and in a few other places where we use panic? It would be more predictable if we did not through panic in the middle of the business logic.

Hey. I totally agree with this. But this doesn't run in runtime and it makes much easier to read the code. This error will finish the program as the panic does.